File: src/main/java/com/github/zxh/classpy/classfile/constant/ConstantFloatInfo.java
package com.github.zxh.classpy.classfile.constant;

/*
CONSTANT_Float_info {
    u1 tag;
    u4 bytes;
}
*/
public class ConstantFloatInfo extends ConstantInfo {

    {
        u4("bytes");
    }
    
    @Override
    protected String loadDesc(ConstantPool pool) {
        float f = Float.intBitsToFloat(super.getUInt("bytes"));
        return Float.toString(f);
    }
    
}

File: src/main/java/com/github/zxh/classpy/classfile/constant/ConstantFloatInfo.kt
package com.github.zxh.classpy.classfile.constant;

/*
CONSTANT_Float_info {
    u1 tag;
    u4 bytes;
}
*/
class ConstantFloatInfo : ConstantInfo() {

    init {
        u4("bytes");
    }
    
    override fun loadDesc(cp: ConstantPool): String {
        val f = java.lang.Float.intBitsToFloat(super.getInt("bytes"));
        return f.toString();
    }
    
}
